pkg/policy/validate.go

```diff
@@ -0,0 +1,119 @@
+package policy
+
+import (
+	"context"
+	"fmt"
+	"path"
+
+	"github.com/cloudquery/cloudquery/pkg/core/database"
+
+	"github.com/cloudquery/cq-provider-sdk/provider/diag"
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgerrcode"
+	"github.com/jackc/pgx/v4"
+	"github.com/rs/zerolog/log"
+	"github.com/thoas/go-funk"
+)
+
+type ValidateRequest struct {
+	// Policy we want to validate
+	Policy *Policy
+	// Directory is where policies reside
+	Directory string
+}
+
+func Validate(ctx context.Context, storage database.Storage, req *ValidateRequest) diag.Diagnostics {
+	log.Info().Str("policy", req.Policy.Name).Str("version", req.Policy.Version()).Str("subPath", req.Policy.SubPolicy()).Msg("preparing to run policy")
+	loadedPolicy, err := Load(ctx, req.Directory, req.Policy)
+	if err != nil {
+		return diag.FromError(err, diag.INTERNAL)
+	}
+	// filter policy based on sub-policy if given
+	filteredPolicy := loadedPolicy.Filter(loadedPolicy.SubPolicy())
+	// always pass root policy identifiers even if they don't exist.
+	return validatePolicy(ctx, storage, &filteredPolicy, loadedPolicy.Identifiers, "")
+}
+
+func validatePolicy(ctx context.Context, storage database.Storage, policy *Policy, identifiers []string, policyPath string) diag.Diagnostics {
+	var diags diag.Diagnostics
+	if len(policy.Identifiers) > 0 {
+		log.Debug().Strs("previous_identifiers", identifiers).Strs("identifiers", policy.Identifiers).Msg("overriding parent policy identifiers")
+		identifiers = policy.Identifiers
+	}
+	if identifiers == nil {
+		diags = diags.Add(diag.FromError(fmt.Errorf("policy %s has no identifiers set", path.Join(policyPath, policy.Name)), diag.USER, diag.WithSeverity(diag.WARNING)))
```

Comment:
```suggestion
		diags = diags.Add(diag.FromError(fmt.Errorf("policy %q has no identifiers set", path.Join(policyPath, policy.Name)), diag.USER, diag.WithSeverity(diag.WARNING)))
```

I suggest using `%q` when referring to user input.